coast/gridded.py

```diff
@@ -0,0 +1,865 @@
+                                                   + ds_dom.e3w_0[1:, :, :])
+            # bottom and below levels
+            e3w_new = e3w_new.where(e3w_new.z_dim < ds_dom.bottom_level, e3t_dt.shift(z_dim=1) + ds_dom.e3w_0)
+            e3_return.append(e3w_new.squeeze())
+
+        return tuple(e3_return)
+
+    def harmonics_combine(self, constituents, components=['x', 'y']):
+        '''
+        Contains a new NEMO object containing combined harmonic information
+        from the original object. 
+        
+        NEMO saves harmonics to individual variables such as M2x, M2y... etc. 
+        This routine will combine these variables (depending on constituents)
+        into a single data array. This new array will have the new dimension
+        'constituent' and a new data coordinate 'constituent_name'.
+        
+        Parameters
+        ----------
+        constituents : List of strings containing constituent names to combine.
+                       The case of these strings should match that used in 
+                       NEMO output. If a constituent is not found, no problem,
+                       it just won't be in the combined dataset.
+        components   : List of strings containing harmonic components to look
+                       for. By default, this looks for the complex components
+                       'x' and 'y'. E.g. if constituents = ['M2'] and
+                       components is left as default, then the routine looks
+                       for ['M2x', and 'M2y']. 
+
+        Returns
+        -------
+        NEMO() object, containing combined harmonic variables in a new dataset.
+        '''
+
+        # Select only the specified constituents. NEMO model harmonics names are
+        # things like "M2x" and "M2y". Ignore current harmonics. Start by constructing
+        # the possible variable names
+        names_x = np.array([cc + components[0] for cc in constituents])
+        names_y = np.array([cc + components[1] for cc in constituents])
+        constituents = np.array(constituents, dtype='str')
+
+        # Compare against names in file
+        var_keys = np.array(list(self.dataset.keys()))
+        indices = [np.where(names_x == ss) for ss in names_x if ss in var_keys]
+        indices = np.array(indices).T.squeeze()
+
+        # Index the possible names to match file names
+        print(indices)
+        names_x = names_x[indices]
+        names_y = names_y[indices]
+        constituents = constituents[indices]
+
+        # Concatenate x and y variables into one array
+        x_arrays = [self.dataset[ss] for ss in names_x]
+        harmonic_x = 'harmonic_' + components[0]
+        x_data = xr.concat(x_arrays, dim='constituent').rename(harmonic_x)
+        y_arrays = [self.dataset[ss] for ss in names_y]
+        harmonic_y = 'harmonic_' + components[1]
+        y_data = xr.concat(y_arrays, dim='constituent').rename(harmonic_y)
+
+        nemo_harmonics = Gridded()
```

Comment:
There might be a problem here. I think it expects a json file. Unit test 1i struggle here, leading to 1j failing.

The aim here is to just combine a bunch of similar gridded objects using xr.merge. However unless an extra config file str is passed then it will not know which type of gridded obj to define.